src/chain_service/faucet.ts

```diff
@@ -152,9 +153,8 @@ export default class Faucet {
   }
 
   private async handleDispatchError(dispatchError: any) {
-    let api = await this.apiManager.getApi();
     if (dispatchError?.isModule) {
-      const decoded = api.api.registry.findMetaError(dispatchError.asModule);
+      const decoded = await this.apiManager.executeApiCall(async (api) => api.registry.findMetaError(dispatchError.asModule));
```

Comment:
Do you think we also need this here? I would assume that even if the api object is outdated and cannot call anymore, querying something from the `.registry` object would still work. It's also not an async operation. I'm fine to keep the changes though, just in case.

Reply:
Mmm I modified it for consistency, but later I added a condition such that only invalid signature errors would reload the object. I doubt querying the registry will ever return such an error, so you are right. Let me modify it to what it was before since it is simpler to read.